### DISCUSSION TITLE
Bridge nodemcu autonomo

### DISCUSSION THREAD

**Discussion by essegi65** [indomus/forum#685] [Hardware & Firmware]
Sto realizzando il bridge zigbee seguendo [questa ](https://indomus.it/progetti/realizzare-un-bridge-gateway-zigbee-mqtt-autonomo-con-nodemcu-e-cc2530-e-affini/)guida indomus. Mi è venuto un dubbio sul firmware Tasmota da utilizzare sul nodemcu, visto che lo tasmotizzo con il programma Nodemcu-pyflasher io pensavo di utilizzare il file tasmota.bin normale (e poi cambiare il template dal browser) o devo andare su tasmota-zigbee.bin?
Grazie in anticipo

**Comment by snakuzzo** [comment]
Se non è cambiato nulla, non puoi fare così. Dovresti scaricare, modificare il codice per abilitare zigbee, compilare e flashare.
Controlla tra i bin disponibili sul sito tasmota prima, magari lo trovi precompilato

**Reply by essegi65** [reply]
Grazie. Avevo controllato sul sito tasmota e gli unici bin che fanno riferimento a zigbee sono: Tasmota-zigbee.bin e tasmota-zibbridge.bin non avendoli mai usati non so se e quale dei 2 fa al caso mio.
Da quello che sono riuscito a capire dal sito tasmota forse è il file tasmota-zigbee.bin

**Reply by snakuzzo** [reply]
Sì...è quello

**Reply by essegi65** [reply]
ok, grazie. Continuo la configurazione seguendo la guida

**Reply by snakuzzo** [reply]
Sì ma considera che la nostra guida è vecchiotta. Il concetto di base rimane, ma lato Tasmota hanno aggiunto diverse cose che in guida non ci sono. 

Leggi la pagina relativa [QUI](https://tasmota.github.io/docs/Zigbee/)

E la pagina dei comandi [QUI](https://tasmota.github.io/docs/Commands/)

**Reply by snakuzzo** [reply]
Inoltre, se non sei pratico, dai un'occhiata a ZHA (anche se non so se puoi usare il nodemcu)

**Reply by essegi65** [reply]
Grazie. Ho appena finito di costruire il tutto. e sembra tutto funzionare sia il nodemcu che il cc2530. Anche lato tasmota da console vedo la verifica come da guida. ancora non ho acquistato device zigbee per fare il paring, appena riesco faccio anche questo.

**Reply by essegi65** [reply]
L'unico dubbio che ho è se uso il bridge collegato ad una presa di corrente normale e collegato via wifi quando configuro zigbee2mqtt come indirizzo di porta cosa devo mettere? l'indirizzo ip?
Forse ho capito, se non ho letto male, nel mio caso non serve installare l'add-on zigbee2mqtt?

**Reply by snakuzzo** [reply]
Eh infatti... che c'entra zigbee2mqtt ??? :D
Mi sa che hai/avevi confuso un po' le cose

**Reply by essegi65** [reply]
eh.... già la bramosia di arrivare a concludere :)

**Reply by essegi65** [reply]
Visto che io sono alle prime armi con zigbeeTasmota qualcuno riesce a dirmi se il componente Bticino L4431C (per controllare i consumi alle prese living now) si riesce ad collegarlo al bridge con esecuzione il firmware Tasmota-zigbee? Prima di acquistarlo vorrei essere sicura della compatibilità.
Grazie

**Reply by snakuzzo** [reply]
Nessuna certezza. Posso dirti che in teoria con Tasmota puoi fare il pairing di qualsiasi device perché poi puoi andare ad interagire con il device andando direttamente ad agire sugli specifici cluster ed endpoint.
Non so se hai letto e studiato già il tutorial Tasmota, ma tocca sbatterci un po' la testa se un device non è gestito lato software per poter mettere a disposizione dell'utente "cose già pronte".
Ti direi... compra, prova ed eventualmente fai un reso.

**Reply by essegi65** [reply]
Si @snakuzzo ho iniziato ma fra che il mio inglese è a livello delle "elementari" e il tempo a disposizione preferivo fare esperienza direttamente provando, magari cercando di fare acquisti mirati.
Quindi cercherò di acquisire il device zigbee e fare test su test. (tanto poi se non riesco scrivo ;) giusto?)

**Reply by snakuzzo** [reply]
Se non hai tempo...lascia perdere questa soluzione ed affidati a zigbee2mqtt.
Sul "se non riesco scrivo" ... no... sbagliato :) anche perché non avendo quei device a disposizione sarebbe un casino darti una mano.

**Reply by essegi65** [reply]
Zigbee2mqtt lo avevo scartato perchè non posso installare il dongle usb sul mio RPI (troppo lontano e i device da comandare sono ad un piano differente) e poi non volevo acquistare "roba" già fatta e mi sono divertito a costruire da me il mio bridge.
Se poi esiste un la possibilità di fare un bridge con usb, per metterlo dove ho bisogno e non attaccato al mio RPI,  e farlo lavorare con Zigbee2Mqtt ben venga.
Era solo una battuta il mio "se non riesco scrivo" :), se non sarà possibile cercherò di cavarmela.

**Reply by snakuzzo** [reply]
Cerca "z-bee duo" sul nostro sito e avrai la risposta

**Reply by essegi65** [reply]
Grazie avevo valutato le soluzioni presenti sul sito tra cui anche z-bee duo. Lo store dove si acquista è sempre senza disponibilità (senza contare che poi sono 38$), avevo provato anche a cercare di acquistare i componenti separati senza riuscirci, anzi se mi date un indicazione dove trovarli ben venga, anche se la spesa non credo vari di molto.

**Reply by snakuzzo** [reply]
La chiavetta è ideata e creata da @gio-dot. Contatta direttamente lui...anche su Telegram (nick @giobin)

**Reply by gio-dot** [reply]
> Grazie avevo valutato le soluzioni presenti sul sito tra cui anche z-bee duo. Lo store dove si acquista è sempre senza disponibilità (senza contare che poi sono 38$), avevo provato anche a cercare di acquistare i componenti separati senza riuscirci, anzi se mi date un indicazione dove trovarli ben venga, anche se la spesa non credo vari di molto.

Per me lascia stare Tasmota.. Semmai affidati ad una soluzione ser2net come la mia. Puoi farla con wemos d1, node mcu etc. però ti ci vuole un coordinator zigbee seriale (non usb) tipo cc2530. Considera però che il cc2530 è obsoleto e sconsigliato per nuove installazioni.
Riguardo la disponibilità della mia mi puoi contattare in pvt. Sullo Store vanno e vengono.

**Reply by essegi65** [reply]
Grazie @Gio-dot, quindi se ho ben capito con nodemcu e cc2530 (ancorché obsoleto) potrei usare zigbee2mqtt?
Questo grazie al fatto di flashare il node con ser2net? anche se non saprei come fare. (io ho gia flashato la cc2530 con il debugger e firmware TexasInst).
Tra l'altro se non ho capito male sul nodemcu potrei anche mettere il firmware tasmota-bridge.bin, aprire la porta di comunicazione e farlo vedere in zigbee2mqtt (serial > 'TCP:indirizzoIP:porta')?
Scusate se insisto con Tasmota ma ho già acquistato hardware e fatto la scatola di assemblaggio, per iniziare volevo non disperdere il tutto

**Reply by gio-dot** [reply]
Esatto. Lascia perdere Tasmota, è pesante per fare la ser2net. Io uso Esp-link oppure Esphome con componente custom ser2net.

**Reply by gio-dot** [reply]
Tasmota, Esp link o Esphome non centrano niente con la scatola.. 😅

**Reply by essegi65** [reply]
Ok grazie mi metto a studiare.
La scatola è fondamentale!....... :):):)

**Reply by essegi65** [reply]
Volevo solo mettere a conoscenza, se potesse magari servire ad altri, del risultato che sono riuscito ad ottenere per creare un proprio bridge autonomo zigbee2mqtt con esp866. (spero che @Gio-dot non me ne voglia)
Non so se posso mettere il link della guida che ho seguito (casomai i moderatori lo toglieranno)che è [questo](https://www.zigbee2mqtt.io/information/connecting_cc2530.html#setting-up-the-esp8266). 
Il firmware, contrariamente a quanto avevo utilizzato all'inizio, da utilizzare NON è tasmota-zigbee ma tasmota-zbbridge e seguendo sia gli schemi di collegamento sia i successivi comandi da console che dice la guida, sono riuscito ad installare su HA il protocollo zigbee2mqtt.
Se a qualcuno necessitasse aiuto sarò ben lieto di farlo. grazie a tutti quelli che mi hanno risposto in questa discussione

**Reply by gio-dot** [reply]
> Volevo solo mettere a conoscenza, se potesse magari servire ad altri, del risultato che sono riuscito ad ottenere per creare un proprio bridge autonomo zigbee2mqtt con esp866. (spero che @Gio-dot non me ne voglia)
> Non so se posso mettere il link della guida che ho seguito (casomai i moderatori lo toglieranno)che è [questo](https://www.zigbee2mqtt.io/information/connecting_cc2530.html#setting-up-the-esp8266).
> Il firmware, contrariamente a quanto avevo utilizzato all'inizio, da utilizzare NON è tasmota-zigbee ma tasmota-zbbridge e seguendo sia gli schemi di collegamento sia i successivi comandi da console che dice la guida, sono riuscito ad installare su HA il protocollo zigbee2mqtt.
> Se a qualcuno necessitasse aiuto sarò ben lieto di farlo. grazie a tutti quelli che mi hanno risposto in questa discussione

Ci mancherebbe. 2 soli appunti: il cc2530/31 è un chip obsoleto e ne è sconsigliato l'utilizzo (sullo stesso sito zigbee2mqtt lo si può vedere).
Il firmware Tasmota per fare la ser2net non è la scelta migliore (anzi dalee mie prove è la peggiore come stabilità e prestazioni): come dicevo sopra molto meglio Esp-link o Esphome.

**Reply by snakuzzo** [reply]
Confermo quello che dice @Gio-dot.
Usa "Esp-Link", non Tasmota

**Reply by essegi65** [reply]
Un aggiornamento, ho provato ad installare zigbee2mqtt v.1.18 e come detto sopra era tutto funzionante.
Ho poi fatto un aggiornamento alla versione 1.21 (eliminando la precedente versione) installando add-on come da guida indomus.
Però dopo la configurazione uguale alla precedente mi rende questo errore nel LOG:

Zigbee2MQTT:error 2021-09-15 21:32:46: Error while starting zigbee-herdsman
Zigbee2MQTT:error 2021-09-15 21:32:46: Failed to start zigbee
Zigbee2MQTT:error 2021-09-15 21:32:46: Check https://www.zigbee2mqtt.io/information/FAQ.html#help-zigbee2mqtt-fails-to-start for possible solutions
Zigbee2MQTT:error 2021-09-15 21:32:46: Exiting...
Zigbee2MQTT:error 2021-09-15 21:32:47: Error: Failed to connect to the adapter (Error: SRSP - SYS - ping after 6000ms)
    at ZStackAdapter.start (/app/node_modules/zigbee-herdsman/src/adapter/z-stack/adapter/zStackAdapter.ts:102:27)
    at Controller.start (/app/node_modules/zigbee-herdsman/src/controller/controller.ts:123:29)
    at Zigbee.start (/app/lib/zigbee.js:66:27)
    at Controller.start (/app/lib/controller.js:110:27)
    at start (/app/index.js:95:5)

sembra il mancato collegamento al server MQTT 
in configurazione ho queste righe:
mqtt:
  base_topic: zigbee2mqtt
  server: mqtt://localhost:1883
  user: mqtt_user
  password: xxxxxxx
serial:
  port: tcp://192.168.x.xx:8899

ovviamente il server MQTT è invece raggiungibile in quanto ho diversi dispostivi sonoff e shelly tutti comandati tramite broker mosquitto
sapete darmi una mano?
grazie
